src/System.Text.Json/tests/Serialization/PolymorphicTests.cs

```diff
@@ -125,6 +129,102 @@ public static void ArrayAsRootObject()
 
             json = JsonSerializer.ToString<object>(ireadonlylist);
             Assert.Equal(ExpectedJson, json);
+
+            Stack<object> stack = new Stack<object>(new List<object> { 1, true, address, null, "foo" });
+            json = JsonSerializer.ToString(stack);
+            Assert.Equal(ReversedExpectedJson, json);
+
+            json = JsonSerializer.ToString<object>(stack);
+            Assert.Equal(ReversedExpectedJson, json);
+
+            Queue<object> queue = new Queue<object>(new List<object> { 1, true, address, null, "foo" });
+            json = JsonSerializer.ToString(queue);
+            Assert.Equal(ExpectedJson, json);
+
+            json = JsonSerializer.ToString<object>(queue);
+            Assert.Equal(ExpectedJson, json);
+
+            HashSet<object> hashset = new HashSet<object>(new List<object> { 1, true, address, null, "foo" });
+            json = JsonSerializer.ToString(hashset);
+            Assert.Equal(ExpectedJson, json);
+
+            json = JsonSerializer.ToString<object>(hashset);
+            Assert.Equal(ExpectedJson, json);
+
+            LinkedList<object> linkedlist = new LinkedList<object>(new List<object> { 1, true, address, null, "foo" });
+            json = JsonSerializer.ToString(linkedlist);
+            Assert.Equal(ExpectedJson, json);
+
+            json = JsonSerializer.ToString<object>(linkedlist);
+            Assert.Equal(ExpectedJson, json);
+
+            IImmutableList<object> iimmutablelist = ImmutableList.CreateRange(new List<object> { 1, true, address, null, "foo" });
+            json = JsonSerializer.ToString(iimmutablelist);
+            Assert.Equal(ExpectedJson, json);
+
+            json = JsonSerializer.ToString<object>(iimmutablelist);
+            Assert.Equal(ExpectedJson, json);
+
+            IImmutableStack<object> iimmutablestack = ImmutableStack.CreateRange(new List<object> { 1, true, address, null, "foo" });
+            json = JsonSerializer.ToString(iimmutablestack);
+            Assert.Equal(ReversedExpectedJson, json);
+
+            json = JsonSerializer.ToString<object>(iimmutablestack);
+            Assert.Equal(ReversedExpectedJson, json);
+
+            IImmutableQueue<object> iimmutablequeue = ImmutableQueue.CreateRange(new List<object> { 1, true, address, null, "foo" });
+            json = JsonSerializer.ToString(iimmutablequeue);
+            Assert.Equal(ExpectedJson, json);
+
+            json = JsonSerializer.ToString<object>(iimmutablequeue);
+            Assert.Equal(ExpectedJson, json);
+
+            IImmutableSet<object> iimmutableset = ImmutableHashSet.CreateRange(new List<object> { 1, true, address, null, "foo" });
+            json = JsonSerializer.ToString(iimmutableset);
+            foreach (string obj in expectedObjects)
```

Comment:
I like this pattern vs. multiple Assert statements. Perhaps add a string extension method?

Reply:
I'll add this to https://github.com/dotnet/corefx/issues/37520.